### ISSUE TITLE
error_logging_for_empty_stations

### ISSUE THREAD

**Issue by MartinRaetz** [RWTH-EBC/AixWeather#18]
**Is your feature request related to a problem? Please describe.**
If weatherdata is pulled and the station contains no weather data it leads to a somehow cryptic error: 
`ValueError: zero-size array to reduction operation fmin which has no identity`

**Describe the solution you'd like**
Let the program run through even without any data, so that at least the meta data can be exported. Here information like "station works until" are present. Such information is also in the station list, but due to different sub measurements per station its not clear whether data is available or not. 

Also check the station list, as the webapp shows another list than the readme.

**Describe alternatives you've considered**
Alternativily at least return the error message "no data"

**Additional context**
Error occurs for e.g. station 433.

**Comment by MartinRaetz** [comment]
# Forecast
Pulling an inactive or non-existing forecast raises a ProcessLookupError. 

```
    try:
        values = next(stations.values.query())
    except Exception as excep:
        raise ProcessLookupError(f"There is no loadable forecast for station {station}") from excep
```

Which is returned by Django as an error page, stating: There is no loadable forecast for station 01223.
This is a good handling of that error if that message is shown even if Django is not in debug mode.
In that case no need for changes.

**Comment by MartinRaetz** [comment]
# DWD historical

- [ ] running non existent stations leads to the error 'STAT_ID', catch this error.

**Comment by MartinRaetz** [comment]
This issue fixes a bug regarding the package versioning of the Python environment, which occurred on some machines and distributions but not on all.

It also makes error messages more descriptive.

**Comment by MartinRaetz** [comment]
> # Forecast
> 
> Pulling an inactive or non-existing forecast raises a ProcessLookupError.
> 
> ```
>     try:
>         values = next(stations.values.query())
>     except Exception as excep:
>         raise ProcessLookupError(f"There is no loadable forecast for station {station}") from excep
> ```
> 
> Which is returned by Django as an error page, stating: There is no loadable forecast for station 10505. In terms of error message this is sufficient in my opinion.
> 
>     * [x]  Error message
> 
> 
> **Searching for the error cause:**
> 
>     * Surprisingly, this error does not occur on my local web app server running with the same main both webapp and aixweather as the VM. Here, it does load the data.
> 
>     * Running the forecast test in aixweather (station: 06710 and 10505) directly triggers another error while unpickling within Wetterdienst: `_pickle.UnpicklingError: invalid load key, '{'.` (PYPI Release Commit)
> 
>     * Running the example in aixweather (station: 10505) triggers the `There is no loadable forecast for station 10505` error. (PYPI Release Commit)
> 
>     * reinstalling python environment using the requirements.py of the webapp of the VM version raises pip error.
> 
> 
> ```
>  (Collecting package metadata (current_repodata.json): ...working... failed
> 
> 
> CondaSSLError: OpenSSL appears to be unavailable on this machine. OpenSSL is required to
> download and install packages.
> 
> Exception: HTTPSConnectionPool(host='repo.anaconda.com', port=443): Max retries exceeded with url: /pkgs/main/win-64/current_repodata.json (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available."))
> ```
> 
> installing AixWeather alone via pip raises an error with Wetterdienst:
> 
> ```
> (test_aixweather_webapp) C:\Users\mre>pip install AixWeather
> Collecting AixWeather
>   Using cached AixWeather-0.1.2.tar.gz (39 kB)
>   Preparing metadata (setup.py) ... error
>   error: subprocess-exited-with-error
> 
>   × python setup.py egg_info did not run successfully.
>   │ exit code: 1
>   ╰─> [54 lines of output]
>       D:\01_Anaconda\envs\test_aixweather_webapp\Lib\site-packages\setuptools\__init__.py:84: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated.
>       !!
> 
>               ********************************************************************************
>               Requirements should be satisfied by a PEP 517 installer.
>               If you are using pip, you can try `pip install --use-pep517`.
>               ********************************************************************************
> 
>       !!
>         dist.fetch_build_eggs(dist.setup_requires)
>       ERROR: Ignored the following versions that require a different python version: 0.29.0 Requires-Python >=3.8,<3.11; 0.30.0 Requires-Python >=3.8,<3.11; 0.30.1 Requires-Python >=3.8,<3.11; 0.31.0 Requires-Python >=3.8,<3.11; 0.31.1 Requires-Python >=3.8,<3.11; 0.32.0 Requires-Python >=3.8,<3.11; 0.32.1 Requires-Python >=3.8,<3.11; 0.32.2 Requires-Python >=3.8,<3.11; 0.32.3 Requires-Python >=3.8,<3.11; 0.32.4 Requires-Python >=3.8,<3.11; 0.33.0 Requires-Python >=3.8,<3.11; 0.34.0 Requires-Python >=3.8,<3.11; 0.35.0 Requires-Python >=3.8,<3.11; 0.36.0 Requires-Python >=3.8,<3.11; 0.37.0 Requires-Python >=3.8,<3.11; 0.38.0 Requires-Python >=3.8,<3.11; 0.39.0 Requires-Python >=3.8,<3.11; 0.40.0 Requires-Python >=3.8,<3.11; 0.41.0 Requires-Python >=3.8,<3.11; 0.41.1 Requires-Python >=3.8,<3.11; 0.42.0 Requires-Python >=3.8,<3.11; 0.42.1 Requires-Python >=3.8,<3.11; 0.43.0 Requires-Python >=3.8,<3.11; 0.44.0 Requires-Python >=3.8,<3.11; 0.45.0 Requires-Python >=3.8,<3.11; 0.45.1 Requires-Python >=3.8,<3.11; 0.45.2 Requires-Python >=3.8,<3.11; 0.46.0 Requires-Python >=3.8,<3.11; 0.47.0 Requires-Python >=3.8,<3.11; 0.47.1 Requires-Python >=3.8,<3.11; 0.48.0 Requires-Python >=3.8,<3.12; 0.49.0 Requires-Python >=3.8,<3.12; 0.50.0 Requires-Python >=3.8,<3.12; 0.51.0 Requires-Python >=3.8,<3.12; 0.52.0 Requires-Python >=3.8,<3.12; 0.53.0 Requires-Python >=3.8,<3.12; 0.54.0 Requires-Python >=3.8,<3.12; 0.54.1 Requires-Python >=3.8,<3.12; 0.55.0 Requires-Python >=3.9,<3.12; 0.55.1 Requires-Python >=3.9,<3.12; 0.56.0 Requires-Python >=3.9,<3.12; 0.56.1 Requires-Python >=3.9,<3.12; 0.56.2 Requires-Python >=3.9,<3.12; 0.57.0 Requires-Python >=3.9,<3.12; 0.57.1 Requires-Python >=3.9,<3.12; 0.58.0 Requires-Python >=3.9,<3.12; 0.58.1 Requires-Python >=3.9,<3.12; 0.59.0 Requires-Python >=3.9,<3.12; 0.59.1 Requires-Python >=3.9,<3.12; 0.59.2 Requires-Python >=3.9,<3.12; 0.59.3 Requires-Python >=3.9,<3.12; 0.60.0 Requires-Python >=3.9,<3.12
>       ERROR: Could not find a version that satisfies the requirement wetterdienst==0.59.0 (from versions: 0.1.0, 0.1.1, 0.2.0, 0.3.0, 0.4.0, 0.5.0, 0.6.0, 0.7.0, 0.8.0, 0.9.0, 0.10.0, 0.10.1, 0.11.0, 0.11.1, 0.12.0, 0.12.1, 0.13.0, 0.14.0, 0.14.1, 0.15.0, 0.16.0, 0.16.1, 0.17.0, 0.18.0, 0.19.0, 0.20.0, 0.20.1, 0.20.2, 0.20.3, 0.20.4, 0.21.0, 0.23.0, 0.24.0, 0.25.0, 0.25.1, 0.26.0, 0.27.0, 0.28.0, 0.61.0, 0.62.0, 0.63.0, 0.64.0, 0.65.0)
>       ERROR: No matching distribution found for wetterdienst==0.59.0
>       Traceback (most recent call last):
>         File "D:\01_Anaconda\envs\test_aixweather_webapp\Lib\site-packages\setuptools\installer.py", line 96, in _fetch_build_egg_no_warn
>           subprocess.check_call(cmd)
>         File "D:\01_Anaconda\envs\test_aixweather_webapp\Lib\subprocess.py", line 413, in check_call
>           raise CalledProcessError(retcode, cmd)
>       subprocess.CalledProcessError: Command '['D:\\01_Anaconda\\envs\\test_aixweather_webapp\\python.exe', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', 'C:\\Users\\mre\\AppData\\Local\\Temp\\tmppp8e3sl2', '--quiet', 'wetterdienst==0.59.0']' returned non-zero exit status 1.
> 
>       The above exception was the direct cause of the following exception:
> 
>       Traceback (most recent call last):
>         File "<string>", line 2, in <module>
>         File "<pip-setuptools-caller>", line 34, in <module>
>         File "C:\Users\mre\AppData\Local\Temp\pip-install-d2ub7zke\aixweather_3027bc5d59f344cc9766edeaa1c237f5\setup.py", line 35, in <module>
>           setuptools.setup(
>         File "D:\01_Anaconda\envs\test_aixweather_webapp\Lib\site-packages\setuptools\__init__.py", line 106, in setup
>           _install_setup_requires(attrs)
>         File "D:\01_Anaconda\envs\test_aixweather_webapp\Lib\site-packages\setuptools\__init__.py", line 79, in _install_setup_requires
>           _fetch_build_eggs(dist)
>         File "D:\01_Anaconda\envs\test_aixweather_webapp\Lib\site-packages\setuptools\__init__.py", line 84, in _fetch_build_eggs
>           dist.fetch_build_eggs(dist.setup_requires)
>         File "D:\01_Anaconda\envs\test_aixweather_webapp\Lib\site-packages\setuptools\dist.py", line 907, in fetch_build_eggs
>           return _fetch_build_eggs(self, requires)
>                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "D:\01_Anaconda\envs\test_aixweather_webapp\Lib\site-packages\setuptools\installer.py", line 38, in _fetch_build_eggs
>           resolved_dists = pkg_resources.working_set.resolve(
>                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "D:\01_Anaconda\envs\test_aixweather_webapp\Lib\site-packages\pkg_resources\__init__.py", line 829, in resolve
>           dist = self._resolve_dist(
>                  ^^^^^^^^^^^^^^^^^^^
>         File "D:\01_Anaconda\envs\test_aixweather_webapp\Lib\site-packages\pkg_resources\__init__.py", line 865, in _resolve_dist
>           dist = best[req.key] = env.best_match(
>                                  ^^^^^^^^^^^^^^^
>         File "D:\01_Anaconda\envs\test_aixweather_webapp\Lib\site-packages\pkg_resources\__init__.py", line 1135, in best_match
>           return self.obtain(req, installer)
>                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "D:\01_Anaconda\envs\test_aixweather_webapp\Lib\site-packages\pkg_resources\__init__.py", line 1147, in obtain
>           return installer(requirement)
>                  ^^^^^^^^^^^^^^^^^^^^^^
>         File "D:\01_Anaconda\envs\test_aixweather_webapp\Lib\site-packages\setuptools\installer.py", line 98, in _fetch_build_egg_no_warn
>           raise DistutilsError(str(e)) from e
>       distutils.errors.DistutilsError: Command '['D:\\01_Anaconda\\envs\\test_aixweather_webapp\\python.exe', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', 'C:\\Users\\mre\\AppData\\Local\\Temp\\tmppp8e3sl2', '--quiet', 'wetterdienst==0.59.0']' returned non-zero exit status 1.
>       [end of output]
> 
>   note: This error originates from a subprocess, and is likely not a problem with pip.
> error: metadata-generation-failed
> 
> × Encountered error while generating package metadata.
> ╰─> See above for output.
> 
> note: This is an issue with the package mentioned above, not pip.
> hint: See above for details.
> 
> (test_aixweather_webapp) C:\Users\mre>
> ```
> 
> installing wetterdienst==0.59.0 directly returned:
> 
> ```
> (test_aixweather_webapp) C:\Users\mre>pip install wetterdienst==0.59.0
> ERROR: Ignored the following versions that require a different python version: 0.29.0 Requires-Python >=3.8,<3.11; 0.30.0 Requires-Python >=3.8,<3.11; 0.30.1 Requires-Python >=3.8,<3.11; 0.31.0 Requires-Python >=3.8,<3.11; 0.31.1 Requires-Python >=3.8,<3.11; 0.32.0 Requires-Python >=3.8,<3.11; 0.32.1 Requires-Python >=3.8,<3.11; 0.32.2 Requires-Python >=3.8,<3.11; 0.32.3 Requires-Python >=3.8,<3.11; 0.32.4 Requires-Python >=3.8,<3.11; 0.33.0 Requires-Python >=3.8,<3.11; 0.34.0 Requires-Python >=3.8,<3.11; 0.35.0 Requires-Python >=3.8,<3.11; 0.36.0 Requires-Python >=3.8,<3.11; 0.37.0 Requires-Python >=3.8,<3.11; 0.38.0 Requires-Python >=3.8,<3.11; 0.39.0 Requires-Python >=3.8,<3.11; 0.40.0 Requires-Python >=3.8,<3.11; 0.41.0 Requires-Python >=3.8,<3.11; 0.41.1 Requires-Python >=3.8,<3.11; 0.42.0 Requires-Python >=3.8,<3.11; 0.42.1 Requires-Python >=3.8,<3.11; 0.43.0 Requires-Python >=3.8,<3.11; 0.44.0 Requires-Python >=3.8,<3.11; 0.45.0 Requires-Python >=3.8,<3.11; 0.45.1 Requires-Python >=3.8,<3.11; 0.45.2 Requires-Python >=3.8,<3.11; 0.46.0 Requires-Python >=3.8,<3.11; 0.47.0 Requires-Python >=3.8,<3.11; 0.47.1 Requires-Python >=3.8,<3.11; 0.48.0 Requires-Python >=3.8,<3.12; 0.49.0 Requires-Python >=3.8,<3.12; 0.50.0 Requires-Python >=3.8,<3.12; 0.51.0 Requires-Python >=3.8,<3.12; 0.52.0 Requires-Python >=3.8,<3.12; 0.53.0 Requires-Python >=3.8,<3.12; 0.54.0 Requires-Python >=3.8,<3.12; 0.54.1 Requires-Python >=3.8,<3.12; 0.55.0 Requires-Python >=3.9,<3.12; 0.55.1 Requires-Python >=3.9,<3.12; 0.56.0 Requires-Python >=3.9,<3.12; 0.56.1 Requires-Python >=3.9,<3.12; 0.56.2 Requires-Python >=3.9,<3.12; 0.57.0 Requires-Python >=3.9,<3.12; 0.57.1 Requires-Python >=3.9,<3.12; 0.58.0 Requires-Python >=3.9,<3.12; 0.58.1 Requires-Python >=3.9,<3.12; 0.59.0 Requires-Python >=3.9,<3.12; 0.59.1 Requires-Python >=3.9,<3.12; 0.59.2 Requires-Python >=3.9,<3.12; 0.59.3 Requires-Python >=3.9,<3.12; 0.60.0 Requires-Python >=3.9,<3.12
> ERROR: Could not find a version that satisfies the requirement wetterdienst==0.59.0 (from versions: 0.1.0, 0.1.1, 0.2.0, 0.3.0, 0.4.0, 0.5.0, 0.6.0, 0.7.0, 0.8.0, 0.9.0, 0.10.0, 0.10.1, 0.11.0, 0.11.1, 0.12.0, 0.12.1, 0.13.0, 0.14.0, 0.14.1, 0.15.0, 0.16.0, 0.16.1, 0.17.0, 0.18.0, 0.19.0, 0.20.0, 0.20.1, 0.20.2, 0.20.3, 0.20.4, 0.21.0, 0.23.0, 0.24.0, 0.25.0, 0.25.1, 0.26.0, 0.27.0, 0.28.0, 0.61.0, 0.62.0, 0.63.0, 0.64.0, 0.65.0)
> ERROR: No matching distribution found for wetterdienst==0.59.0
> ```
> 
> Creating a new environment and installing the requirements.py of the AixWeather Functional Repo manually and without version information worked until installing `pip install wetterdienst`, tried `pip install wetterdienst==0.65.0` with same result:
> 
> ```
>   note: This error originates from a subprocess, and is likely not a problem with pip.
>   ERROR: Failed building wheel for pyarrow
> Failed to build aiohttp pyarrow
> ERROR: Could not build wheels for aiohttp, pyarrow, which is required to install pyproject.toml-based projects
> ```
> 
> `pip install wetterdienst==0.59.0` results in the `ERROR: No matching distribution found for wetterdienst==0.59.0`.
> 
> Running the tests at the same commit results in:
> 
>     * via CI: tests pass
> 
>     * via local machine: test results in known error `_pickle.UnpicklingError: invalid load key, '{'.`
> 
>     * via local machine with "old" environment of the aixweather-webapp its running through
> 
> 
> @Jun-Jiang-92 installed the aixweather functional repo via pycharm venv and the setup.py. The installation worked fine. Also the tests including the forecast-test ran through without error.
> 
> _Findings 1_ The reason must be within the Python environment. Ideas: Order of installation does affect the version of primary packages which are used in different secondary packages like wetterdienst.
> 
> _Findings 2_ Updating `pip install fsspec==2023.9.2` at the previously failing environment solves the problem.
> 
>     * [x]  completely rearrange the requirements.txt and setup.py
> 
>     * [x]  fix respective deprecation warnings

This issue still remains on the VM.

**Comment by leonw97** [comment]
I did not encounter any errors during the installation and local execution of the AixWeather WebApp. To do this, I create Python environments with anaconda. I have tested this for environments of Python versions 3.9 to 3.11. I then install the required content via the requirements.txt using pip (version 21.0.1).

After creating the environment, I can start the local WebApp via Pycharm and download valid weather data under the DWD-forecast tab for station 10505. Windows 10 is used as the operating system.
